### DISCUSSION TITLE
Ideas for additional phone options

### DISCUSSION THREAD

**Discussion by matthewmayer** [faker-js/faker#2883] [General]
This discussion is to keep track of suggestions for further improvements to the `faker.phone.number()` method.

In v9 this is already being changed to use a "style" parameter.
https://next.fakerjs.dev/guide/upgrading.html#phone-number-style-replaces-explicit-format

However there may be use-cases which are not well covered by the eexisting "styles"

**Comment by matthewmayer** [comment]
Suggested at https://github.com/faker-js/faker/issues/1542#issuecomment-2042098388 - ability to generate cellphone/mobile numbers only

**Comment by matthewmayer** [comment]
Suggested at https://github.com/faker-js/faker/issues/1542#issuecomment-2097931945 - what can we learn from the FFaker Ruby gem implementation?

https://www.rubydoc.info/github/ffaker/ffaker/FFaker/PhoneNumber

**Comment by matthewmayer** [comment]
Support "special" numbers - https://github.com/faker-js/faker/issues/2839

**Comment by matthewmayer** [comment]
More strict checking for valid NANP (+1) phone numbers - https://github.com/faker-js/faker/issues/2303

**Comment by dumbasPL** [comment]
Cellphone number generation (Less strict variant). For counties where it is impossible to easily distinguish and generate cellphone and landline numbers, generating any valid number would be sufficient. Reasoning: when a service that only has basic phone numbers checks, for counties where it's not trivial to tell them apart, it's ok to generate anything since the other side won't be able to tell the difference anyway.

**Comment by cabello** [comment]
In my recent case faker generated a toll free phone number with extension `+1 800 555 1234 x4567` and Stripe prevented the payment intent from being created because they limit phone numbers to a max of 20 chars, the quick non-code fix was easy, I removed the extension code from that test record and got unblocked.

It would be nice that I could specify in the options that a) I don't want `1 800` or business looking numbers and b) I don't want extensions added to it.

Something like:
```javascript
faker.phone.number({ style: "international", tollFree: false, extension: false })
```

Thank you. 🙇

**Comment by rafaelbeckel** [comment]
The current international option generates numbers ﻿in the E.123 format. We should also have an E.124 option. Some APIs validate over it, and E.123 does not pass validation for all cases.
